README.md

```diff
@@ -385,15 +425,267 @@ import { parseHeaders } 'imaginary-http-parser'
 // imagine that parseHeaders has the following signature:
 // parseHeaders(raw: string): Result<SomeKeyValueMap, ParseError>
 
-const promise = parseHeaders(rawHeader)
+const asyncRes = parseHeaders(rawHeader)
   .map(headerKvMap => headerKvMap.Authorization)
   .asyncMap(findUserInDatabase)
 ```
 
-Note that in the above example if `parseHeaders` returns an `Err` then `.map` and `.asyncMap` will not be invoked, and `promise` variable will contain a `Err` inside of the promise.
+Note that in the above example if `parseHeaders` returns an `Err` then `.map` and `.asyncMap` will not be invoked, and `asyncRes` variable will resolve to an `Err` when turned into a `Result` using `await` or `.then()`.
+
+---
+
+### `okAsync`
+
+Constructs an `Ok` variant of `ResultAsync`
+
+**Signature:**
+
+```typescript
+okAsync<T, E>(value:  T): ResultAsync<T, E>
+```
+
+**Example:**
+
+```typescript
+import { okAsync } from 'neverthrow'
+
+const myResultAsync = okAsync({ myData: 'test' }) // instance of `ResultAsync`
+
+const myResult = await myResultAsync // instance of `Ok`
+
+myResult.isOk() // true
+myResult.isErr() // false
+```
+
+---
+
+### `errAsync`
+
+Constructs an `Err` variant of `ResultAsync`
+
+**Signature:**
+
+```typescript
+errAsync<T, E>(err:  E):  ResultAsync<T, E>
+```
+
+**Example:**
+
+```typescript
+import { errAsync } from 'neverthrow'
+
+const myResultAsync = errAsync('Oh nooo') // instance of `ResultAsync`
+
+const myResult = await myResultAsync // instance of `Err`
+
+myResult.isOk() // false
+myResult.isErr() // true
+```
 
 ---
+
+### `ResultAsync.fromPromise` (method)
+
+Transforms a `Promise<T>` into a `ResultAsync<T, E>`.
+
+The second argument handles the rejection case of the promise. If it is ommited, the code might throw.
```

Comment:
```suggestion
The second argument handles the rejection case of the promise. If it is ommited, **the code might throw** because `neverthrow` does not know if the promise you are passing to `fromPromise` has any promise rejection logic associated to it (via a `.catch` method call or `catch (err) {}` block).
```